src/main/java/com/adyen/model/terminal/SaleToAcquirerDataModel.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+    public String getTenderOption() {
+        return tenderOption;
+    }
+
+    public void setTenderOption(String tenderOption) {
+        this.tenderOption = tenderOption;
+    }
+
+    public Map<String, String> getAdditionalData() {
+        return additionalData;
+    }
+
+    public void setAdditionalData(Map<String, String> additionalData) {
+        this.additionalData = additionalData;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        SaleToAcquirerDataModel that = (SaleToAcquirerDataModel) o;
+        return Objects.equals(metadata, that.metadata) &&
+                Objects.equals(shopperEmail, that.shopperEmail) &&
+                Objects.equals(shopperReference, that.shopperReference) &&
+                Objects.equals(recurringContract, that.recurringContract) &&
+                Objects.equals(shopperStatement, that.shopperStatement) &&
+                Objects.equals(recurringDetailName, that.recurringDetailName) &&
+                Objects.equals(recurringTokenService, that.recurringTokenService) &&
+                Objects.equals(store, that.store) &&
+                Objects.equals(merchantAccount, that.merchantAccount) &&
+                Objects.equals(currency, that.currency) &&
+                Objects.equals(applicationInfo, that.applicationInfo) &&
+                Objects.equals(tenderOption, that.tenderOption) &&
+                Objects.equals(additionalData, that.additionalData);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(metadata, shopperEmail, shopperReference, recurringContract, shopperStatement, recurringDetailName, recurringTokenService, store, merchantAccount, currency, applicationInfo, tenderOption, additionalData);
+    }
+
+    @Override
+    public String toString() {
+        return "SaleToAcquirerDataModel{" +
+                "metadata=" + metadata +
+                ", shopperEmail='" + shopperEmail + '\'' +
+                ", shopperReference='" + shopperReference + '\'' +
+                ", recurringContract='" + recurringContract + '\'' +
+                ", shopperStatement='" + shopperStatement + '\'' +
+                ", recurringDetailName='" + recurringDetailName + '\'' +
+                ", recurringTokenService='" + recurringTokenService + '\'' +
+                ", store='" + store + '\'' +
+                ", merchantAccount='" + merchantAccount + '\'' +
+                ", currency='" + currency + '\'' +
+                ", applicationInfo=" + applicationInfo +
+                ", tenderOption='" + tenderOption + '\'' +
+                ", additionalData=" + additionalData +
+                '}';
+    }
+
+    public String toBase64() {
+        String json = PRETTY_PRINT_GSON.toJson(this);
+        return new String(Base64.encodeBase64(json.getBytes()));
```

Comment:
json.getBytes may return in null pointer exception.

Reply:
I am wondering when this could happen. this is always an object with application info so it should always contain data.

Reply:
"this" won't be null but it is possible that conversion to json has some problem and returns in empty json string.